portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,10 +34,13 @@
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private static int maxComments = 10;
```

Comment:
Something to consider: what happens when you deploy this code and two people (or more) start using your site and leaving comments. Can they independently choose how many comments they would like to see?

Reply:
If they use the input box to enter the number of comments they want to see on the page without reloading, they can choose the number of comments independently. But when they want to reload the page, the value of the limit might be affected because the maxComments value is stored in the datastore. But I haven't came up with a feasible way to fix this since I have to store the number of maxComments somewhere when the page is refreshed.

Reply:
Yep. One thing, though: the maxComments value isn't stored in Datastore, but rather as an instance variable of this DataServlet running somewhere. It will get reset any time the servlet is restarted.

Reply:
So, for a solution, you could imagine tracking this value for each user separately, but then you'd have to figure out a way to tell one user from another. That's possible, but out-of-scope for this PR (and probably this week).

However, this is a case where designing your web server to be stateless can help us. (Stateless meaning the server itself doesn't track any state, like it does with `maxComments` currently.) We can do this by making the user *tell* us their `maxComments` preference on every request, which we can do with query parameters in GET requests. In that way, the server doesn't have to remember anything about previous requests, and its job becomes easier because it is always told exactly what to return.

To fix the reload problem you mentioned, we can use JavaScript. In addition to making `fetchCommentsWithInputLimit` send the user-provided `maxComments` every time, we also update the JavaScript to fix page reloads. There is a way to adjust the brower's current URL from JS (https://developer.mozilla.org/en-US/docs/Web/API/History_API; see `history.pushState`); using this, we can set maxComments in the query string, and have our JavaScript check there on page-load to see if we should update the default value for the input.

Does that make sense? It's ok to do all that in a new PR, if you like.

Reply:
Yeah I totally understand that it is better for my website to be stateless, which can prevent the situation that multiple users concurrently modify the status of the server. I think I will make the website stateless in a new PR. But personally I actually really like the idea that tracks the `maxComments` value for each user separately. Maybe I will try it out in a future PR.

Reply:
Sounds good to try something different in a new PR. I recommend starting with statelessness, as implementing user association is a bigger topic and might come up later on.